### PR TITLE
[CN-650]-Add e2e test sub group PR builder

### PR DESCRIPTION
**Pull request by cheels**
## Description

- Updated the Kind Nightly Tests Workflow so that from now all the tests will be split between 10 KinD nodes, which decreases the entire test run from 50m to 17m. 
- Add this split run to the PR builder
- The tests will be constantly shuffled and equally distributed between suites/nodes
- Test report collects all XMLs from all nodes and merges them into one report
- Removed the 'New Relic' link from the report
- Remove Before and After test records from the report
- Added Hazelcast favicon to test report pages

## User Impact

PR builds and Kind Nightly Test becomes much faster then before

### CODE REVIEW

**Review by mtyazici — COMMENTED** [review]
I see there are multiple conflict files, should we rebase the changes?

File: .github/workflows/nightly-e2e-kind.yaml [hunk 1]
```diff
@@ -29,7 +91,7 @@ jobs:
         with:
           path: |
             ~/.cache/go-build
-            ~/go/pkg/mod
+            ~/go/pkg/mod: ~/go/pkg/mod
```
[the remark this answers is not in the document]
Comment (cheels) [reply .github/workflows/nightly-e2e-kind.yaml]: I removed nightly-kind tests since we will run kind tests in PR builder

**Review by hasancelik — APPROVED** [review]

File: .github/scripts/utils.sh [hunk 2]
```diff
@@ -193,6 +194,129 @@ wait_for_instance_restarted()
...
+update_test_files()
+{
+      local WORKFLOW_ID=$1
+      local CLUSTER_NAME=$2
+      cd allure-history/$WORKFLOW_ID/${GITHUB_RUN_NUMBER}/data/test-cases
+      local GRAFANA_BASE_URL="https://hazelcastoperator.grafana.net"
+      local BEGIN_TIME=$(date +%s000 -d "- 3 hours")
+      local END_TIME=$(date +%s000 -d "+ 1 hours")
+      for i in $(ls); do
+          cat <<< $(jq -e 'del(.testStage.steps[] | select(has("name") and (.name | select(contains("CR_ID")|not)) and (.name | select(contains("Text")|not))))
```
Comment (dzeromski-hazelcast) [inline .github/scripts/utils.sh:280]: WOW :superhero: but can we rewrite it in golang? :stuck_out_tongue:

Comment (cheels) [reply .github/scripts/utils.sh:279]: @dzeromski-hazelcast Probably in the future, not this time :)

**Review by dzeromski-hazelcast — APPROVED** [review]

**Pull request merged**